### PR TITLE
Carefully release ByteBuf when processing WebSocketFrame

### PR DESCRIPTION
**Pull request by kachayev**
Trying to catch #428. Left a few notes/comments in the code.

@ztellman Seems more reasonable now, but it obviously requires a double-check.

### CODE REVIEW

**Comment by shilder** [comment]
I'm not sure how this request changes Binary frames processing - it's essentialy the same.

Bug #428 was introduced in https://github.com/ztellman/aleph/commit/64a8819d - it adds `.fireChannelRead` without acquiring buffer, hence double free.

**Comment by kachayev** [comment]
@shilder This is fixed now, earlier `release` was done as a finalizer. Right now there's no "global" try-catch for all use cases, meaning you don't need `acquire` anything when simply passing to the next handler.

**Comment by shilder** [comment]
Yes, i've seen that, but you mentioned in issue

> It looks like acquire for a message and message body is a bit messed up when dealing with binary messages, which leads to the error on finally....

And i wanted to point out that binary messages are handled just fine.

Also you don't need `if-not (instance? WebSocketFrame msg)` clause - `:else` branch of cond handles this case just fine.

**Comment by kachayev** [comment]
> Also you don't need if-not (instance? WebSocketFrame msg) clause - :else branch of cond handles this case just fine.

That's a habbit of checking class before doing

```[^WebSocketFrame msg msg]```.

**Comment by shilder** [comment]
> That's a habbit of checking class before doing
> 
> `[^WebSocketFrame msg msg]`.

Yes, i understand that, but every clause in `cond` has `instance?` check, so that's unnecessary in this case

**Comment by kachayev** [comment]
@shilder Get rid both of `if` branching and type hint for a message (as we have type hint per each frame individually).

**Comment by shilder** [comment]
Thanks, looks cleaner now, and yes, that `finally` block was confusing. I guess we can get rid of it in `websocket-client-handler` too (it will be more complicated though)

File: src/aleph/http/server.clj [hunk 1]
```diff
@@ -534,33 +534,41 @@
...
-                   (instance? CloseWebSocketFrame msg)
-                   (.close handshaker ch (netty/acquire msg))
-
-                   :else
-                   (.fireChannelRead ctx msg)))))
-           (finally
-             (netty/release msg)))))]))
+         (let [ch (.channel ctx)]
+           (cond
+             (instance? TextWebSocketFrame msg)
```
Comment (gsnewmark) [inline src/aleph/http/server.clj]: @kachayev you could use [`(condp instance? msg ...)`](https://clojuredocs.org/clojure.core/condp) to simplify code further.

Comment (shilder) [reply src/aleph/http/server.clj]: That can be a little excessive :) 

My goal was to get rid of second `fireChannelRead`, and I don't think `condp` is used anywhere in aleph, so just for consistency i would prefer to stick with `cond`

Comment (kachayev) [reply src/aleph/http/server.clj]: If my memory doesn't fail me... `condp` was used for handling websocket frames on the client side. I personally found it harder to read and reason about, but that might be just my personal preference :)

**Comment by ztellman** [comment]
@kachayev also broken by the ping functionality

**Comment by kachayev** [comment]
@ztellman Okay... merged this one. I'll do #422 after this one is merged, I think they will have more conflicts.

**Pull request merged**